libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionText.java

```diff
@@ -1,157 +1,79 @@
 package com.mapbox.services.android.navigation.ui.v5.instruction;
 
+import android.text.TextUtils;
+
+import com.mapbox.directions.v5.models.BannerInstructions;
 import com.mapbox.directions.v5.models.LegStep;
 import com.mapbox.services.android.navigation.v5.utils.abbreviation.StringAbbreviator;
-import com.mapbox.services.utils.TextUtils;
 
-import java.util.Arrays;
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ARRIVE;
 
 public class InstructionText {
 
-  private TextFields stepTextFields;
+  private String primaryText;
+  private String secondaryText;
   private double stepDistance;
+  private double distanceAlongStep;
 
-  public InstructionText(LegStep step) {
+  public InstructionText(LegStep step, double distanceAlongStep) {
+    primaryText = "";
+    secondaryText = "";
     stepDistance = step.distance();
-    stepTextFields = buildInstructionText(step);
+    if (distanceAlongStep > 0d) {
+      this.distanceAlongStep = distanceAlongStep;
+    }
+    buildInstructionText(step);
   }
 
   public String getPrimaryText() {
-    return StringAbbreviator.abbreviate(stepTextFields.primaryText);
+    return StringAbbreviator.abbreviate(primaryText);
```

Comment:
Shouldn't we be able to get away with not having to abbreviate anymore since this text is coming from the server?